lib/chef/knife/bootstrap.rb

```diff
@@ -588,13 +598,30 @@ def register_client
       def perform_bootstrap(remote_bootstrap_script_path)
         ui.info("Bootstrapping #{ui.color(server_name, :bold)}")
         cmd = bootstrap_command(remote_bootstrap_script_path)
+        bootstrap_run_command(cmd)
+      end
+
+      # Actual bootstrap command to be run on the node.
+      # Handles recursive calls if su USER failed to authenticate.
+      def bootstrap_run_command(cmd)
         r = connection.run_command(cmd) do |data|
           ui.msg("#{ui.color(" [#{connection.hostname}]", :cyan)} #{data}")
         end
         if r.exit_status != 0
-          ui.error("The following error occurred on #{server_name}:")
-          ui.error(r.stderr)
-          exit 1
+          stderr = (r.stderr + r.stdout).strip
+
+          if stderr.match?("su: Authentication failure")
```

Comment:
This is probably going to be O/S dependent and will break on platforms that aren't AIX.

I know that I made a comment on the train side that the patch there didn't seem sufficient to do it train-side, but this code is now making me reconsider and that most of this code should actually get pushed into train itself.  That way these kinds of strings can be tuned to the exact operating system (and can also be leveraged by inspec, etc)

Reply:
As of now, I have tested these changes with the following nodes and everything working fine.

| Workstation                | Workstation OS |  Node            |  Matcher Data    |
|----------------------------|:--------------:|-----------------:|-----------------:|
| Chef Infra Client: 16.5.60 | Ubuntu 18.04   | Ubuntu 16.04     | Password:        |
| Chef Infra Client: 16.5.60 | Ubuntu 18.04   | Ubuntu 20.04     | Password:        |
| Chef Infra Client: 16.5.60 | Ubuntu 18.04   | Centos 8         | Password:        |
| Chef Infra Client: 16.5.60 | Ubuntu 18.04   |   AIX 7.1        | root's Password: |
| Chef Infra Client: 16.5.60 | Ubuntu 18.04   |   AIX 7.2        | root's Password: |